File: platform/engine/registry_test.go
package engine

import (
	"testing"
)

type mockAdapter struct {
	name string
	tier Tier
}

func (m *mockAdapter) Name() string    { return m.name }
func (m *mockAdapter) Tier() Tier      { return m.tier }
func (m *mockAdapter) Execute(prompt string) (*RawResponse, error) {
	return &RawResponse{EngineName: m.name, Content: "mock"}, nil
}

func TestRegistry_RegisterAndGet(t *testing.T) {
	r := NewRegistry()
	a := &mockAdapter{name: "test", tier: TierDirect}
	r.Register(a)

	got := r.Get("test")
	if got == nil {
		t.Fatal("Get('test') nil döndü")
	}
	if got.Name() != "test" {
		t.Errorf("beklenen 'test', gerçek %s", got.Name())
	}
}

func TestRegistry_GetUnknown(t *testing.T) {
	r := NewRegistry()
	got := r.Get("unknown")
	if got != nil {
		t.Error("bilinmeyen adapter nil dönmeli")
	}
}

func TestRegistry_List(t *testing.T) {
	r := NewRegistry()
	r.Register(&mockAdapter{name: "a"})
	r.Register(&mockAdapter{name: "b"})
	r.Register(&mockAdapter{name: "c"})

	names := r.List()
	if len(names) != 3 {
		t.Errorf("beklenen 3, gerçek %d", len(names))
	}
}

func TestRegistry_Count(t *testing.T) {
	r := NewRegistry()
	r.Register(&mockAdapter{name: "x"})
	r.Register(&mockAdapter{name: "y"})
	if r.Count() != 2 {
		t.Errorf("beklenen 2, gerçek %d", r.Count())
	}
}

func TestRegistry_EmptyCount(t *testing.T) {
	r := NewRegistry()
	if r.Count() != 0 {
		t.Errorf("beklenen 0, gerçek %d", r.Count())
	}
}

File: platform/java/src/test/java/dev/geolens/engine/RegistryTest.java
package dev.geolens.engine;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNull;

/** Go {@code registry_test.go} portu. */
class RegistryTest {

    private static Adapter mock(String name, Tier tier) {
        return new Adapter() {
            @Override
            public String name() {
                return name;
            }

            @Override
            public Tier tier() {
                return tier;
            }

            @Override
            public RawResponse execute(String prompt) {
                return new RawResponse(name, "", "mock", null, false, tier, "", "");
            }

            @Override
            public Adapter withContext(String tenantId, String workspaceId) {
                return this;
            }
        };
    }

    @Test
    void registerAndGet() {
        Registry r = new Registry();
        Adapter a = mock("test", Tier.DIRECT);
        r.register(a);

        Adapter got = r.get("test");
        assertEquals("test", got.name());
    }

    @Test
    void getUnknown() {
        Registry r = new Registry();
        assertNull(r.get("unknown"));
    }

    @Test
    void list() {
        Registry r = new Registry();
        r.register(mock("a", Tier.DIRECT));
        r.register(mock("b", Tier.DIRECT));
        r.register(mock("c", Tier.DIRECT));

        assertEquals(3, r.list().size());
    }

    @Test
    void count() {
        Registry r = new Registry();
        r.register(mock("x", Tier.DIRECT));
        r.register(mock("y", Tier.DIRECT));
        assertEquals(2, r.count());
    }

    @Test
    void emptyCount() {
        Registry r = new Registry();
        assertEquals(0, r.count());
    }
}